controllers/metal3machine_controller_test.go

```diff
@@ -455,6 +455,85 @@ var _ = Describe("Metal3Machine manager", func() {
 		),
 	)
 
+	type TestCaseM3DToBMM struct {
+		Data          *infrav1.Metal3Data
+		ExpectRequest bool
+	}
+
+	DescribeTable("Metal3Data To Metal3Machines tests",
+		func(tc TestCaseM3DToBMM) {
+			r := Metal3MachineReconciler{}
+			obj := handler.MapObject{
+				Object: tc.Data,
+			}
+			reqs := r.Metal3DataToMetal3Machines(obj)
+
+			if tc.ExpectRequest {
+				Expect(len(reqs)).To(Equal(1), "Expected 1 request, found %d", len(reqs))
+
+				req := reqs[0]
+				Expect(req.NamespacedName.Name).To(Equal(tc.Data.Spec.Metal3Machine.Name),
+					"Expected name %s, found %s", tc.Data.Spec.Metal3Machine.Name, req.NamespacedName.Name)
+				if tc.Data.Spec.Metal3Machine.Namespace == "" {
+					Expect(req.NamespacedName.Namespace).To(Equal(tc.Data.Namespace),
+						"Expected namespace %s, found %s", tc.Data.Namespace, req.NamespacedName.Namespace)
+				} else {
+					Expect(req.NamespacedName.Namespace).To(Equal(tc.Data.Spec.Metal3Machine.Namespace),
+						"Expected namespace %s, found %s", tc.Data.Spec.Metal3Machine.Namespace, req.NamespacedName.Namespace)
+				}
+
+			} else {
+				Expect(len(reqs)).To(Equal(0), "Expected 0 request, found %d", len(reqs))
+
+			}
+		},
+		Entry("No Metal3Machine in Spec",
+			TestCaseM3DToBMM{
+				Data: &infrav1.Metal3Data{
+					ObjectMeta: metav1.ObjectMeta{
```

Comment:
ObjectMeta struct could be defined separately and re-used. Less repetition and shorter TC Entry statements.

Reply:
done, thanks